src/App.tsx

```diff
@@ -17,59 +17,139 @@ const goodsFromServer = [
 
 enum SortType {
   NONE,
-  ALPABET,
+  ALPHABET,
   LENGTH,
 }
 
-// Use this function in the render method
-function getReorderedGoods(
+const getReorderedGoods = (
   goods: string[],
   sortType: SortType,
   isReversed: boolean,
-) {
-  // Not to mutate the original array
+) => {
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // ...
+  if (sortType !== SortType.NONE) {
+    visibleGoods.sort((f1, f2) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return f1.localeCompare(f2);
+
+        case SortType.LENGTH:
+          return f1.length - f2.length;
+
+        default:
+          return 0;
+      }
+    });
```

Comment:
It is good practice when you can avoid nested ifs. Code will be cleaner and will have better readability
```suggestion
      switch (sortType) {
        case SortType.ALPHABET:
          return visibleGoods.sort((f1, f2) => f1.localeCompare(f2));
        case SortType.LENGTH:
          return visibleGoods.sort((f1, f2) => f1.length - f2.length);
        case SortType.NONE:
        default:
          return 0;
      }
```